### PR TITLE
Cached remove

### PR DESCRIPTION
**Pull request by zarinfam**
Add invalidate (remove) capability to Cached annotation:
https://github.com/playframework/playframework/issues/4019

### CODE REVIEW

File: framework/src/play-cache/src/main/java/play/cache/CachedAction.java [hunk 1]
```diff
@@ -8,26 +8,32 @@
...
                 promise = delegate.call(ctx);
-                promise.onRedeem(new F.Callback<Result>() {
-                    @Override
-                    public void invoke(Result result) throws Throwable {
-                        Cache.set(key, result, duration);
-                    }
-                });
-            } else {
-                promise = F.Promise.pure(result);
+            }else {
```
Comment (benmccann) [inline framework/src/play-cache/src/main/java/play/cache/CachedAction.java]: Need a space after the close brace

**Comment by jroper** [comment]
What's the difference between this and #4021?  We'll also need tests and documentation about this in the Play manual if we do end up accepting it.

**Comment by zarinfam** [comment]
#4021 is another approach and defined a new annotation instead of add new property to Cached annotation. I prefer that solution.

**Comment by benmccann** [comment]
@zarinfam I'm going to close this given that you said you prefer the other solution

**Pull request closed without merging**